src/array/ops/filter.rs

```diff
@@ -73,37 +76,20 @@ impl crate::datatypes::PythonArray {
     }
 }
 
-impl DateArray {
-    pub fn filter(&self, mask: &BooleanArray) -> DaftResult<Self> {
-        let new_array = self.physical.filter(mask)?;
-        Ok(Self::new(self.field.clone(), new_array))
-    }
+macro_rules! impl_logical_array_filter {
```

Comment:
Since all of these are the same, what do you think of putting it into the logicalarray impl https://github.com/Eventual-Inc/Daft/blob/main/src/datatypes/logical.rs#L15